src/pages/instances/InstanceDetailHeader.tsx

```diff
@@ -125,20 +128,34 @@ const InstanceDetailHeader: FC<Props> = ({
           ) : null
         }
         controls={
-          instance ? (
+          isMediumScreen() ? (
+            <InstanceDetailHeaderDropdown
+              instance={instance}
+              project={project}
+              isLoading={isLoading}
+            />
+          ) : (
             <>
-              {isClustered ? (
-                <MigrateInstanceBtn instance={instance} project={project} />
+              {" "}
+              {instance ? (
+                <>
+                  {isClustered ? (
+                    <MigrateInstanceBtn instance={instance} project={project} />
+                  ) : null}
+                  <CreateImageFromInstanceBtn
+                    key="publish"
+                    instance={instance}
+                  />
+                  <DuplicateInstanceBtn
+                    key="duplicate"
+                    instance={instance}
+                    isLoading={isLoading}
+                  />
+                  <DeleteInstanceBtn key="delete" instance={instance} />
+                </>
```

Comment:
Probably best to move the buttons and all the logic to switch between dropdown and buttons into a dedicated `InstanceDetailActions` component. Maybe we can merge it with the `InstanceDetailHeaderDropdown`.

I think we can define an array of `actions = []` to which we push the buttons (depending on their conditions). Reuse that array for the context menu and for the button rendering. So we don't define the actions and their conditions twice. That would be hard to maintain in the long run and create an extra need to keep the two consistent. Best to rely on a single source of truth for it.